hazelcast/src/main/java/com/hazelcast/map/impl/LazyEntryView.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.core.EntryView;
+import com.hazelcast.internal.serialization.SerializationService;
+
+/**
+ * LazyEntryView is an implementation of {@link EntryView} and also it is
+ * writable. It is lazy because you may intialise it with serialized formats
+ * of key and value and it will deserialise only if {@link #getKey()} or
+ * {@link #getValue()} are invoked.
+ * If the provided key and/or value are in deserialised format, the methods
+ * do not do any additional deserialisation and only return the provided
+ * instances.
+ *
+ * @param <K> the type of key.
+ * @param <V> the type of value.
+ */
+class LazyEntryView<K, V> implements EntryView<K, V> {
```

Comment:
a simple test would be good.

Reply:
Not sure if we should keep it or remove it.

Reply:
if it is not used, we can remove?

Reply:
OK, will remove.